R/table_parts.R

```diff
@@ -272,6 +272,11 @@ tab_spanner <- function(data,
   # of the `grp_labels` list
   grp_labels[column_names] <- label
 
+  # Ensure that the label classes are preserved
+  for (group_label_i in seq(grp_labels[column_names])) {
+    class(grp_labels[column_names][[group_label_i]]) <- class(label)
+  }
```

Comment:
Hmmm, this is necessary because `grp_labels[columns_names] <- label` automatically drops the class? That's a bit scary--that's an easy mistake to make! Would it help to have a version of `rep_len` that preserves the class, and assign that to `grp_labels[column_names]`?

Also, what's the type of `grp_labels[column_names]`, anyway? Shouldn't it be a list of lists?

Reply:
This is now rewritten to produce a `labels` list of lists of length `grp_labels[column_names]`. With that, the assignment of `grp_labels[column_names] <- labels` produces a list of lists (as before).